api/v1alpha1/gitrepository_types.go

```diff
@@ -0,0 +1,66 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitRepositorySpec struct {
+	Source GitRepositorySource `json:"source,omitempty"`
```

Comment:
How do we see the lifecycle of this resource? One thing that bugs me about the current implementation is this coupling of out of cluster state with resources that live inside the cluster and it looks like we're going to have to answer a similar question here:

* If the CLI is stopped, should this resource still exist?
* If so, then if the CLI is restarted should it resume syncing this gitrepository?
* If we do that, do we need to perform some guarding so multiple git repositories arent fighting to write to the same destination?
* What if someone runs multiple instances of the CLI?

Reply:
I think these questions apply to all resources we create so let's open an issue and talk about it. I agree we need to handle created resources better because currently we don't handle lifecycles of our resources at all. We don't even provide a way to delete created clusters.